src/arguments/CoreMessage.ts

```diff
@@ -0,0 +1,17 @@
+import type { PieceContext } from '@sapphire/pieces';
+import type { DMChannel, Message, NewsChannel, TextChannel } from 'discord.js';
+import { Argument, ArgumentContext, AsyncArgumentResult } from '../lib/structures/Argument';
+
+export class CoreArgument extends Argument<Message> {
+	private readonly snowflakeRegex = /^\d{17,19}$/;
+
+	public constructor(context: PieceContext) {
+		super(context, { name: 'message' });
+	}
+
+	public async run(argument: string, context: ArgumentContext): AsyncArgumentResult<Message> {
+		const channel = (context.channel as DMChannel | NewsChannel | TextChannel | undefined) ?? context.message.channel;
```

Comment:
I wasn't really satisfied with this. Do you suggest something better I could do?

Reply:
I could extend `ArgumentContext` to provide an optional channel, but if I were to, should I export it? If so, I'd probably have to extend it in `Argument.ts` and export it, but I'm not sure if that's really the right place to put it... that said, it would make this look nicer, because then it would be something like:
```ts
public async run(argument: string, context: FittingNameHere): AsyncArgumentResult<Message> {
    const channel = context.channel ?? context.message.channel;
    const message = this.snowflakeRegex.test(argument) ? await channel.messages.fetch(argument).catch(() => null) : null;
    // yada yada 
}
```